controller-server/src/main/java/com/yahoo/vespa/hosted/controller/LockedTenant.java

```diff
@@ -123,23 +126,44 @@ public User with(Contact contact) {
     public static class Cloud extends LockedTenant {
 
         private final BillingInfo billingInfo;
+        private final BiMap<String, Principal> pemDeveloperKeys;
 
-        private Cloud(TenantName name, BillingInfo billingInfo) {
+        private Cloud(TenantName name, BillingInfo billingInfo, BiMap<String, Principal> pemDeveloperKeys) {
             super(name);
             this.billingInfo = billingInfo;
+            this.pemDeveloperKeys = pemDeveloperKeys;
         }
 
         private Cloud(CloudTenant tenant) {
-            this(tenant.name(), tenant.billingInfo());
+            this(tenant.name(), tenant.billingInfo(), tenant.pemDeveloperKeys());
         }
 
         @Override
         public CloudTenant get() {
-            return new CloudTenant(name, billingInfo);
+            return new CloudTenant(name, billingInfo, pemDeveloperKeys);
         }
 
         public Cloud with(BillingInfo billingInfo) {
-            return new Cloud(name, billingInfo);
+            return new Cloud(name, billingInfo, pemDeveloperKeys);
+        }
+
+        public Cloud withPemDeveloperKey(String pemKey, Principal principal) {
+            ImmutableBiMap.Builder<String, Principal> keys = ImmutableBiMap.builder();
```

Comment:
Nit: It's generally better to have the constructor guarantee immutability, rather than individual methods.

Reply:
I agree. I didn't realise the `HashBiMap` also preserves iteration order, but I see it does. Will change in a later PR.